### DISCUSSION TITLE
Event not wokring

### DISCUSSION THREAD

**Discussion by Nuhel** [livewire/livewire#6178] [7. Bugs (archived)] [answered]
### Livewire version

3.12.3

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

Attach a event listener in component class.
Like: 

```php
#[On('post-created')]
public function onPostCreated($title)
{
    dd('onPostCreated');
}
```

try it to dispatch from view using js

```
<div>
    Knowing others is intelligence; knowing yourself is true wisdom.

    <button class="p-4 rounded" @click="()=>{
        alert('on-click')
        $dispatch('post-created')
    }">Dispatch</button>
</div>
```

Nb: alert is working, but dispatch is not working. and with Filament v3.0

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PhiloNL** [comment]
Can you try:

```js
$wire.$dispatch
```

https://livewire.laravel.com/docs/javascript#the-wire-object

**Reply by Nuhel** [reply]
Same result.

**Comment by romansh** [comment]
A little silly notice, but is there not a missing semicolon at the end?
```js
alert('on-click')
```

**Reply by Nuhel** [reply]
js dosent require semicolon.

**Comment by N1ebieski** [comment]
Maybe because you require a parameter $title:

```php
public function onPostCreated($title)
```

but you don't pass it:

```js
$dispatch('post-created')
```

**Reply by Nuhel** [reply]
not the issue, i tried without that.

**Reply by citricguy** [reply]
Both of these worked for me:

`<button @click="$dispatch('post-created', { title: 'Post Title' })">Button3</button>`

You could of course do this as well:

```
<div>
        Knowing others is intelligence; knowing yourself is true wisdom.

        <button class="p-4 rounded" @click="()=>{
        alert('on-click')
        $dispatch('post-created', { title: 'Post Title' })
    }">Dispatch</button>
</div>
```

**Reply by citricguy** [reply]
You could also remove the required $title variable and not pass a title.

i.e.,

```
#[On('post-created')]
public function onPostCreated()
{
    dd('onPostCreated');
}
```

**Comment by XternalSoft** [comment]
hello, same issue for me, from php side.
You can see this demo repository 
https://github.com/XternalSoft/livewire3-dispatch-bug

**Reply by XternalSoft** [reply, marked as answer]
I had forgotten ...

`use Livewire\Attributes\On;`

**Reply by bernhardh** [reply]
Thank you so much! This is really weird, because it doesn't throw any error and isn't documented well!

**Comment by joaopalopes24** [comment]
@Nuhel 

I am with this bug.

In Blade:
```
<button x-on:click="$dispatch('test', { title: 'Post Title' })">
    Show Dialog
</button>
```

In Component:
```
#[On('test')]
public function test(array $options): void
{
    dd($options);
}
```

Error: **Unable to resolve dependency [Parameter #0 [ <required> array $options ]]**

**Reply by XternalSoft** [reply]
Try

```
<button x-on:click="$dispatch('test', title: 'Post Title' )">
    Show Dialog
</button>
```
And
```
#[On('test')]
public function test(string $title): void
{
    dd($title);
}
```

**Reply by joaopalopes24** [reply]
It worked! But I needed to put second parameter in an array.

```
<button x-on:click="$dispatch('test', { title: 'Post Title' })">
    Show Dialog
</button>
```

**Comment by devront-it** [comment]
Your livewire version 3.12.3 does not exist. Do you mean 2.12.3? If so, this approach won't work at all...

**Comment by puklipo** [comment]
1. Clear view cache `php artisan view:clear`
2. Remove alpine.js from `resources/js/app.js` and re build assets `npm run build`
